`ClassNotFoundException` when not using discovery

### Expected Behavior

If user does not want to use discovery (and hence does not have it on the classpath), I'd expect the gRPC server to start without issues.

### Actual Behaviour

The gRPC embedded server cannot start due to the following exception:
```
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [io.micronaut.grpc.server.GrpcEmbeddedServer] could not be loaded: io/micronaut/discovery/cloud/ComputeInstanceMetadataResolver
	at io.micronaut.context.DefaultBeanContext$BeanDefinitionProducer.getDefinition(DefaultBeanContext.java:4244)
	at io.micronaut.context.DefaultBeanContext.collectBeanCandidates(DefaultBeanContext.java:2160)
	at io.micronaut.context.DefaultBeanContext.findBeanCandidates(DefaultBeanContext.java:2135)
	...
	...
Caused by: java.lang.NoClassDefFoundError: io/micronaut/discovery/cloud/ComputeInstanceMetadataResolver
	at io.micronaut.grpc.server.$GrpcEmbeddedServer$Definition.<clinit>(Unknown Source)
	at io.micronaut.grpc.server.$GrpcEmbeddedServer$Definition$Reference.load(Unknown Source)
	at io.micronaut.context.AbstractInitializableBeanDefinitionReference.load(AbstractInitializableBeanDefinitionReference.java:178)
	at io.micronaut.context.DefaultBeanContext$BeanDefinitionProducer.getDefinition(DefaultBeanContext.java:4239)
	... 58 more
Caused by: java.lang.ClassNotFoundException: io.micronaut.discovery.cloud.ComputeInstanceMetadataResolver
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 62 more
```

As a workaround, we can add discovery-core to runtime dependencies.

### Steps To Reproduce

Run `./gradlew run`

### Environment Information

JDK21

### Example Application

https://github.com/mcanalesmayo/micronaut-grpc-bug-no-class-def-found

### Version

4.3.4

--- a/grpc-server-runtime/build.gradle
+++ b/grpc-server-runtime/build.gradle
@@ -14,12 +14,12 @@ dependencies {
     api libs.managed.grpc.stub
 
     compileOnly libs.managed.grpc.services
-    compileOnly mn.micronaut.discovery.core
     compileOnly mn.micronaut.management
     compileOnly libs.graal.svm
 
     implementation libs.grpc.netty
     implementation mn.micronaut.buffer.netty
+    implementation mn.micronaut.discovery.core
 
     testCompileOnly libs.javax.annotation.api
 

--- a/grpc-server-runtime/src/main/java/io/micronaut/grpc/server/GrpcEmbeddedServer.java
+++ b/grpc-server-runtime/src/main/java/io/micronaut/grpc/server/GrpcEmbeddedServer.java
@@ -63,7 +63,10 @@ import static io.micronaut.core.io.socket.SocketUtils.LOCALHOST;
 @Singleton
 @Secondary
 @Named(GrpcServerConfiguration.PREFIX)
-@Requires(classes = ServerBuilder.class)
+@Requires(classes = {
+    ServerBuilder.class,
+    ServiceInstance.class
+})
 @Requires(property = GrpcServerConfiguration.ENABLED, value = StringUtils.TRUE, defaultValue = StringUtils.TRUE)
 public class GrpcEmbeddedServer implements EmbeddedServer {
 

